src/utils/fmt.zig

```diff
@@ -1,38 +1,240 @@
 const std = @import("std");
 const sig = @import("../lib.zig");
 
-inline fn maxArg(comptime field: std.builtin.Type.StructField) field.type {
-    return switch (@typeInfo(field.type)) {
-        .ComptimeInt => sig.utils.types.defaultValue(field) orelse @compileError("comptime_int field without default value has no max value"),
-        .Int => std.math.maxInt(field.type),
-        else => @compileError("[Argument '" ++ field.name ++ "'] Unsupported type: " ++ @typeName(field.type)),
-    };
-}
-
-pub inline fn maxArgs(comptime Args: type) Args {
-    comptime {
-        var max_args: Args = undefined;
-        for (@typeInfo(Args).Struct.fields) |field| {
-            const ptr = &@field(max_args, field.name);
-            ptr.* = maxArg(field);
-        }
-        return max_args;
-    }
-}
-
+/// Returns the maximum length applicable for the format string and `Args` tuple,
+/// such that it would be the equivalent to the length of the bounded array returned
+/// by `boundedFmt`.
+/// ```zig
+/// try expectEqual("255-1".len, boundedLen("{d}-{d}", struct { u8, u1 }));
```

Comment:
kinda interesting you went with a test as an example for the docs - just thinking out loud but maybe we can think of tests as additional documentation - and use that as motivation to keep them relatively close to the methods in the code 🤔

no need to change anything just thinking out loud

Reply:
Yeah, tests are a pretty natural way of expressing "examples". The main issue is that real tests don't currently appear on documentation generated by autodocs. But maybe that's okay, since autodocs does at least link to the source, wherein "documentatitive tests" could be naturally found - and there's always the possibility the tooling will just improve and eventually cover this sort of use case.

But indeed, I hear you and I'm with you.